packages/react-combobox/Spec.md

```diff
@@ -2,62 +2,515 @@
-
-- Decide whether to use **native element** or folow **ARIA** and provide reasons
-- Identify the **[ARIA](https://www.w3.org/TR/wai-aria-practices-1.2/) pattern** and, if the component is listed there, follow its specification as possible.
-- Identify accessibility **variants**, the `role` ([ARIA roles](https://www.w3.org/TR/wai-aria-1.1/#role_definitions)) of the component, its `slots` and `aria-*` props.
-- Describe the **keyboard navigation**: Tab Oder and Arrow Key Navigation. Describe any other keyboard **shortcuts** used
-- Specify texts for **state change announcements** - [ARIA live regions
-  ](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) (number of available items in dropdown, error messages, confirmations, ...)
-- Identify UI parts that appear on **hover or focus** and specify keyboard and screen reader interaction with them
-- List cases when **focus** needs to be **trapped** in sections of the UI (for dialogs and popups or for hierarchical navigation)
-- List cases when **focus** needs to be **moved programatically** (if parts of the UI are appearing/disappearing or other cases)
+While accessibility is built into the Combobox as much as possible, there will always be bugs due to quirks in platform and screen reader support. The most robustly accessible selection component will always be `@fluentui/react-select`, since it uses the native `<select>` element.
+
+Most Combobox-specific accessibility considerations are built in to the rest of the spec, but this section addresses additional concerns that don't fit into any other category, or are worth calling out in more detail.
+
+### Using an inline popup
+
+The inline vs. portal behavior of the listbox popup has a particularly large impact on Combobox accessibility because keyboard focus does not enter the popup. This, coupled with the lack of support for `aria-owns` in Safari, means that iOS VoiceOver users will not be able to access the options unless they either swipe to the end of the DOM, or use touch exploration to try to find the popup on the screen.
+
+If `inline` is set to true, the listbox will follow the trigger in the DOM, and swipe access will work.
+
+The other scenario where `inline` is important is if the Combobox is used inside a modal of any sort. VoiceOver will not allow the cursor to leave a modal, so if the listbox is _not_ rendered inline, it will be fully impossible for an iOS VoiceOver user to use the combobox.
+
+### Semantic structure
+
+The Fluent Combobox uses the [ARIA 1.2 combobox pattern](https://www.w3.org/TR/wai-aria-1.2/#combobox), which differs significantly from ARIA 1.1. The ARIA 1.2 pattern has [better practical support](https://www.24a11y.com/2019/select-your-poison-part-2/), and as of writing, the 1.2 spec is headed towards Candidate Recommendation.
```

Comment:
it would be good to talk to the browser vendors or to the screen reader vendors to improve narration around the aria-controls link. Basically, as input is linked to the listbox, screen reader should be able to announce the number of available options, when it changes in the listbox. This was historically done by a live narration and is therefore tricky

Reply:
That'd be good to raise for sure. I think it may be a bit more complicated than just `aria-controls`, since some decisions about what page changes to announce are a combination of SR vendor verbosity decisions and access via the API. My unsubstantiated guess is that combobox patterns have been shifting so much that support for finer details like this have kind of been lagging behind. Hopefully pushing harder for support can change that.

How would you feel about me noting down the lack of option # change announcements in the "Known Issues" section for now? Ideally we can use that section in the future as a list of things to push for more support on in the future.

Reply:
I added it to the known issues list👍